lib/Doctrine/Inflector/Inflect.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Inflector;
+
+use InvalidArgumentException;
+use function lcfirst;
+use function preg_replace;
+use function sprintf;
+use function str_replace;
+use function strtolower;
+use function ucwords;
+
+class Inflect implements InflectInterface
+{
+    /** @var Singularizer */
+    private $singularizer;
+
+    /** @var Pluralizer */
+    private $pluralizer;
+
+    public function __construct(Singularizer $singularizer, Pluralizer $pluralizer)
+    {
+        $this->singularizer = $singularizer;
+        $this->pluralizer   = $pluralizer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function tableize(string $word) : string
+    {
+        return strtolower(preg_replace('~(?<=\\w)([A-Z])~', '_$1', $word));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function classify(string $word) : string
+    {
+        return str_replace([' ', '_', '-'], '', ucwords($word, ' _-'));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function camelize(string $word) : string
+    {
+        return lcfirst($this->classify($word));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function ucwords(string $string, string $delimiters = " \n\t\r\0\x0B-") : string
+    {
+        return ucwords($string, $delimiters);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset() : void
+    {
+        $this->singularizer->reset();
+        $this->pluralizer->reset();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addRules(string $type, array $rules, bool $reset = false) : void
+    {
+        switch ($type) {
```

Comment:
Instead of having this, why don't we just have `addSingularRule(s)` and `addPluralRule(s)`?